### PR TITLE
Allporncomic ripper

### PR DESCRIPTION
**Pull request by cyian-1756**
# Category

This change is exactly one of the following (please change `[ ]` to `[x]`) to indicate which:
* [ ] a bug fix (Fix #...)
* [X] a new Ripper
* [ ] a refactoring
* [ ] a style change/fix
* [ ] a new feature

# Description

Added a ripper and test for allporncomic.com

# Testing

Required verification:
* [ ] I've verified that there are no regressions in `mvn test` (there are no new failures or errors).
* [x] I've verified that this change works as intended.
  * [ ] Downloads all relevant content.
  * [ ] Downloads content from multiple pages (as necessary or appropriate).
  * [ ] Saves content at reasonable file names (e.g. page titles or content IDs) to help easily browse downloaded content.
* [x] I've verified that this change did not break existing functionality (especially in the Ripper I modified).

Optional but recommended:
* [x] I've added a unit test to cover my change.

### CODE REVIEW

File: src/main/java/com/rarchives/ripme/ripper/rippers/AllporncomicRipper.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Override
+    public String getGID(URL url) throws MalformedURLException {
+        Pattern p = Pattern.compile("https?://allporncomic.com/porncomic/([a-zA-Z0-9_-]+)/([a-zA-Z0-9_-]+)");
+        Matcher m = p.matcher(url.toExternalForm());
+        if (m.matches()) {
+            return m.group(1) + "_" + m.group(2);
+        }
+        throw new MalformedURLException("Expected cfake URL format: " +
+                "cfake.com/picture/MODEL/ID - got " + url + " instead");
```
Comment (kittenparry) [inline src/main/java/com/rarchives/ripme/ripper/rippers/AllporncomicRipper.java]: You forgot to edit this part when copying and pasting from the original values.

Comment (kittenparry) [reply src/main/java/com/rarchives/ripme/ripper/rippers/AllporncomicRipper.java]: Didn't realise you were cyian-1756. I'm sure you were getting there.

Comment (cyian-1756) [reply src/main/java/com/rarchives/ripme/ripper/rippers/AllporncomicRipper.java]: Totally missed that, thanks for pointing it out

Comment (kittenparry) [reply src/main/java/com/rarchives/ripme/ripper/rippers/AllporncomicRipper.java]: Not a problem at all. Not trying to snatch the pull request or anything, but if you've time maybe could you check #1362? Without that the ripper I've added doesn't function on half of the pages. I wanted to send a PM but I guess GitHub doesn't have that.

**Comment by cyian-1756** [comment]
The test failed for an unrelated reason (Looks like we have to update the version of java we test with) so I'm merging this

**Pull request merged**